libgstc/libgstc_socket.c

```diff
@@ -106,28 +121,51 @@ gstc_socket_new (const char *address, const unsigned int port,
 GstcStatus
 gstc_socket_send (GstcSocket * self, const char *request, char **response)
 {
+  GstcStatus ret;
+
   gstc_assert_and_ret_val (NULL != self, GSTC_NULL_ARGUMENT);
   gstc_assert_and_ret_val (NULL != request, GSTC_NULL_ARGUMENT);
   gstc_assert_and_ret_val (NULL != response, GSTC_NULL_ARGUMENT);
 
+  if (!self->keep_connection_open) {
```

Comment:
Same exact pattern as in `gstc_socket_new`. Refactor.

Reply:
Pattern is no longer the same since `gstc_socket_new` always opens the socket and closes it in case the `keep_connection_open` is false. This function only deals with opening-closing logic/error handling in case the flag is set to true